openshift/catalog-sync/pkg/check/catalog.go

```diff
@@ -0,0 +1,91 @@
+package check
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+// AllCatalogChecks returns a list of all catalog checks to be performed.
+func AllCatalogChecks() []CatalogCheck {
+	return []CatalogCheck{
+		CatalogHasValidMetadata(),
+		CatalogHasValidMetadataForChannelHeads(),
```

Comment:
Sorry, I think we flew past each other in yesterday's review I left in this area.

The metadata check should be:
1. All channel heads have `olm.csv.metadata`
2. _Nothing_ has `olm.bundle.object`

We have no opinion (yet) about whether non-channel heads have `olm.csv.metadata`

Reply:
Hi @joelanford 

OK. I am changing for that. However, that does not seems the check that we have in : https://github.com/konflux-ci/build-definitions/blob/main/task/fbc-validation/0.1/fbc-validation.yaml#L236-L253

(Code shared ^ when we did our first meeting)

Reply:
Neverless it is done as requested now 👍 So, addressed.